wd.html

```diff
@@ -444,6 +444,276 @@ <h3>Get Title</h3>
+
+#Click the link which opens in a new window
+driver.find_element(link: 'new window').click
+
+#Wait for the new window or tab
+wait.until { driver.window_handles.length == 2 }
+
+#Loop through until we find a new window handle
+driver.window_handles.each do |handle|
+    if handle != original_window
+        driver.switch_to.window handle
+        break
+    end
+end
+
+#Wait for the new tab to finish loading content
+wait.until { driver.title == 'Selenium documentation'}
+ </code>
+ <code class=javascript>//Store the ID of the original window
+const originalWindow = await driver.getWindowHandle();
+
+//Check we don't have other windows open already
+assert((await driver.getAllWindowHandles()).length === 1);
+
+//Click the link which opens in a new window
+await driver.findElement(By.linkText('new window')).click();
+
+//Wait for the new window or tab
+await driver.wait(function() {
+    return driver.getAllWindowHandles().then(function(windows) {
+        return windows.length === 2;
+    });
+}, 10000);
+
+//Loop through until we find a new window handle
+const windows = (await driver.getAllWindowHandles());
+for (let i = 0; i < windows.length; i++) {
+    if(windows[i]!==originalWindow) {
+        await driver.switchTo().window(windows[i]);
+    }
+}
+
+//Wait for the new tab to finish loading content
+await driver.wait(until.titleIs('Selenium documentation'), 10000);</code>
+</pre>
+
+<h3>Closing a window or tab</h3>
+<p>When you are finished with a window or tab <em>and</em> it is not the
+ last window or tab open in your browser, you should close it and switch
+ back to the window you were using previously.  Assuming you followed the
+ code sample in the previous section you will have the previous window
+ handle stored in a variable. Put this together and you will get:</p>
+<pre>
+ <code class=java>//Close the tab or window
+driver.close();
+
+//Switch back to the old tab or window
+driver.switchTo().window(originalWindow);</code>
+ <code class=python>#Close the tab or window
+driver.close()
+
+#Switch back to the old tab or window
+driver.switch_to.window(original_window)</code>
+ <code class=cs>//Close the tab or window
+driver.Close();
+
+//Switch back to the old tab or window
+driver.SwitchTo().Window(originalWindow);</code>
+ <code class=ruby>#Close the tab or window
+driver.close
+
+#Switch back to the old tab or window
+driver.switch_to.window original_window</code>
+ <code class=javascript>//Close the tab or window
+await driver.close();
+
+//Switch back to the old tab or window
+await driver.switchTo().window(originalWindow);
+ </code>
+</pre>
+<p>Forgetting to switch back to another window handle after closing a
+ window will leave WebDriver executing on the now closed page, and will
+ trigger a <strong>No Such Window Exception</strong>.</p>
+<p>If you didn't store
+ your previous window handle, you can alternatively switch back to the
+ first handle using:</p>
+<pre>
```

Comment:
Selenium makes no guarantee of window handle order, you'd be better off holding the original window handle in a var and switching to it rather than relying on handles[0] to work

Reply:
My thinking here is that it will get you back to a valid window handle if you've just closed a window, avoid a `NoSuchWindow` exception.  But perhaps it is better not to be explicit here and skip this suggestion.